source/usage-examples/findOne.txt

```diff
@@ -3,3 +3,95 @@ Find a Document
 ===============
 
 .. default-domain:: mongodb
+
+You can retrieve a single document from your collection with the
+``*Collection.FindOne()`` method. 
+
+..
+   method
+
+The ``FindOne()`` method takes a query document that specifies the
+subset of your collection that you would like to retrieve a document from. If you
+pass the ``FindOne()`` method an empty document, MongoDB will consider your
+entire collection when retrieving your document. For more information on documents
+in the MongoDB Go driver, see our
+:doc:`guide on data formats </fundamentals/data-formats>`.
+
+You must pass the ``FindOne()`` method a ``context.Context`` interface type
+value. For more information on the ``context.Context`` interface and the
+``context`` package, see our :doc:`guide on context </fundamentals/context>`.
+
+..
+   options
+
+You can specify options that modify the behavior of your ``FindOne()`` method.
+One example of an option is **sort**, which orders the subset of your collection
+that matches your query before the first document is retrieved. Another example
+of an option is a **projection**, which specifies or restricts the fields in
+your returned document. You can specify additional options by passing the
+``FindOne()`` method a pointer to a ``FindOneOptions`` struct. You
+can configure a ``FindOneOptions`` struct by using setter methods such as the
+``*FindOneOptions.SetProjection()`` method.
+
+..
+   return
```

Comment:
This only focuses on the `Decode` method of `SingleResult`. It may be good to mention `DecodeBytes` and `Err` as well. It may also be useful to put the SingleResult docs in a common location as it's returned from a multiple functions.

Reply:
Thanks for the note. I believe we generally do not propose multiple solutions to problems in out "Usage Example" documentation. We generally wait to propose alternative solutions and provide in depth background for driver features until out "Fundamentals" sections. 

I am not sure what you mean by "SingleResult" docs in a common location. Do you mean within this page, or within the entirety of the Go Driver documentation?

I have directed the DevEd team to this comment to ask if deferring this edit until writing Fundamentals sections is appropriate. I'll follow up in this thread with their response.

Reply:
Thanks for following up Kevin. Yes, this section is not meant to fully describe `SingleResult`. [Usage Examples](https://docs.mongodb.com/drivers/node/current/usage-examples/) are intended to provide "convenient starting points" where as Fundamentals provide information on "core concepts".

Apologies for not following up in this thread sooner. While we try to consistently structure our docs, we also try to be flexible and make compromises or amend structure when it could benefit learners. So, while generally we provide one convenient solution to a problem in a usage example (intended as a How-To for common use cases), we are always open to suggestions.

If you are interested in learning more about the structure of Driver's documentation, [this presentation](https://docs.google.com/presentation/d/1YRBdZ3cEWj6UX23dH91OkRL5BY2j68Zzw_0fyAr5NLI/edit#slide=id.g8d9441a72b_1_999) contains a lot of helpful information on the background of the project and reasoning behind organization.

Reply:
That sounds good, and thank you for the additional context!